src/bokeh/core/property/either.py

```diff
@@ -81,20 +81,24 @@ def __init__(self, type_param0: TypeOrInst[Property[Any]], *type_params: TypeOrI
         super().__init__(type_param0, *type_params, default=default, help=help)
         for tp in self.type_params:
             self.alternatives.extend(tp.alternatives)
+        self.valids = []
 
     def transform(self, value: Any) -> Any:
-        for param in self.type_params:
+        for i, param in enumerate(self.type_params):
             try:
-                return param.transform(value)
+                # if valid was not executed, or all valids failed - transform one by one
+                # else: run transform on the first param that pass validation
+                if len(self.valids)==0 or not any(self.valids) or self.valids[i]: 
+                    return param.transform(value)
             except ValueError:
                 pass
 
         raise ValueError(f"Could not transform {value!r}")
 
     def validate(self, value: Any, detail: bool = True) -> None:
         super().validate(value, detail)
-
-        if any(param.is_valid(value) for param in self.type_params):
+        self.valids = [param.is_valid(value) for param in self.type_params]
```

Comment:
I think if we want to efficiently store the first valid param, without excess computation, we could use this:
```python
self.valid = next(p for for p in self.type_params if p.is_valid(value), None)
```
if there are no valid params then the value of `self.valid` will be None

Reply:
Hi, thanks for the comment, but sometime it can happen that 2 pass validation, and we will try to run first the first one that pass validation, and if it does not work, the second one that pass validation.
e.g.
I can sometime see that `self.valid=[False, False, True, False, True]`
So in the case, we want to try to run the 3rd transform, and if it does not work the last transform.
What do you think?

Reply:
I think that if the the third option returns valid, but then subsequently fails to transform, that there must be a bug somwhere else. A passing validation check is supposed to mean exactly that the property _will_ accept the value. If it does not, something else is wrong. Can you provide a concrete test case to demonstrate the situation you describe?

Reply:
Yeah, I need to find the edge case again, i'll get back to you. It also relates to your question below:
"Question: if self.valid is not None, can self.valid.transform fail? If so we might try...except: pass around the first conditional,  ..."